app/flux/components/Checkout.react.js

```diff
@@ -0,0 +1,96 @@
+var React = require('react');
+
+var { State } = require('react-router');
+var StoreAPI = require('../API/StoreAPI');
+
+var UserStore = require('../stores/UserStore');
+
+class Checkout extends React.Component{
+    constructor (props){
+        super(props);
+
+        // Only lifecycle methods receive the inheritance from React.Component.
+        this._onChange = this._onChange.bind(this);
+        
+        this.state = {
+            stripe: null,
+            training: null,
+            user: UserStore.user        
+        };
+    }
+
+    componentDidMount (){
+        UserStore.addChangeListener(this._onChange);
+
+        if (!this.state.user && UserStore.isPending){
+            setTimeout(() => {
```

Comment:
This seems really hacky and brittle, this should be handling dispatched events and not using setTimeout.